bundles/org.openhab.binding.amazonechocontrol/src/main/java/org/openhab/binding/amazonechocontrol/internal/Connection.java

```diff
@@ -1553,87 +1558,97 @@ public void executeSequenceCommand(@Nullable Device device, String command,
     }
 
     private void executeSequenceNode(Device[] devices, JsonObject nodeToExecute) {
-        if (devices.length == 1 && groups.values().stream().anyMatch(queueObject -> queueObject.queueRunning.get())
-                || devices.length > 1
-                        && singles.values().stream().anyMatch(queueObject -> queueObject.queueRunning.get())) {
-            if (singleGroupTimer != null) {
-                singleGroupTimer.cancel(true);
-                singleGroupTimer = null;
+        QueueObject queueObject = new QueueObject();
+        queueObject.devices = devices;
+        queueObject.nodeToExecute = nodeToExecute;
+        String serial = "";
+        for (Device device : devices) {
+            if (!this.devices.containsKey(device.serialNumber)) {
+                this.devices.put(device.serialNumber, new LinkedBlockingQueue<>());
             }
-            singleGroupTimer = scheduler.schedule(() -> executeSequenceNode(devices, nodeToExecute), 500,
-                    TimeUnit.MILLISECONDS);
-
-            return;
+            this.devices.get(device.serialNumber).offer(queueObject);
+            serial = serial + device.serialNumber + " ";
         }
+        logger.debug("added {} device {}", queueObject.hashCode(), serial);
+    }
 
-        if (devices.length == 1) {
-            if (!singles.containsKey(devices[0])) {
-                singles.put(devices[0], new QueueObject());
-            }
-            singles.get(devices[0]).queue.add(nodeToExecute);
-        } else {
-            if (!groups.containsKey(devices[0])) {
-                groups.put(devices[0], new QueueObject());
+    private void handleExecuteSequenceNode() {
+        if (devicesRunning.compareAndSet(false, true)) {
```

Comment:
Judging from how you are using this, please change `devicesRunning` to a Lock and use tryLock instead.